### PR TITLE
Add Helpers Functions for v0.6

### PR DESCRIPTION
**Pull request by rauljordan**
This is part of #3, this PR adds the `pack, merkleize, mixInLength, mixInType` helper functions for SSZ

**Benches With Around 256Kb of Data**
```go
BenchmarkPack-4               	    2000	   1117806 ns/op
BenchmarkMerkleize-4          	     100	  14708158 ns/op
BenchmarkIsPowerTwo-4         	100000000	      13.8 ns/op
```

### CODE REVIEW

File: hash_cache.go [hunk 1]
```diff
@@ -1,33 +1,12 @@
...
-		Help: "The number of hash requests that aren't present in the cache.",
-	})
-	hashCacheHit = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "ssz_hash_cache_hit",
-		Help: "The number of hash requests that are present in the cache.",
-	})
-	hashCacheSize = promauto.NewGauge(prometheus.GaugeOpts{
-		Name: "ssz_hash_cache_size",
-		Help: "The number of hashes in the block cache",
-	})
```
Comment (prestonvanloon) [inline hash_cache.go]: Please put this back. Why move it?

File: helpers.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package ssz
+
+import "github.com/prysmaticlabs/prysm/shared/hashutil"
```
Comment (prestonvanloon) [inline helpers.go]: Why do you need something from Prysm? Let’s avoid this because it is a circular dep for Prysm to depend on this project

**Comment by prestonvanloon** [comment]
Tests? Looks like some methods are incomplete

File: hash_cache.go (hunk 1 above)
Comment (nisdas) [reply hash_cache.go]: the prometheus metrics look better in this file, imo

File: helpers.go (hunk 2 above)
Comment (nisdas) [reply helpers.go]: we already have a hash function in the ssz package. Lets use that, although it needs to be changed to sha256 from keccak256

Comment (prestonvanloon) [reply helpers.go]: https://github.com/prysmaticlabs/go-ssz/blob/2e84733edbac32aca6d47feafc4441e43b10047f/hash.go#L44

File: helpers.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+// Given a Merkle root root and a type_index type_index ("uint256" little-endian serialization)
+// return hash(root + type_index).
+func mixInType(root [32]byte, typeIndex []byte) [32]byte {
+	return Hash(append(root[:], typeIndex...))
+}
+
+// fast verification to check if an number if a power of two.
+func isPowerTwo(num int) bool {
```
Comment (prestonvanloon) [inline helpers.go]: Consider copying this: https://github.com/prysmaticlabs/prysm/blob/f801bfedcf39dfa6e367b920d043cd58fa5c0175/shared/mathutil/math_helper.go#L32

It is much more performant

**Pull request merged**